PatternPal/PatternPal.Extension/PatternPal.ExtensionPackage.cs

```diff
@@ -86,17 +86,23 @@ public bool DoLogData
             CancellationToken cancellationToken,
             IProgress< ServiceProgressData > progress)
         {
-            // NOTE: This code is never called!
 
             // When initialized asynchronously, the current thread may be a background thread at this point.
             // Do any initialization that requires the UI thread after switching to the UI thread.
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-            DTE dte = (DTE)await GetServiceAsync(typeof( DTE ));
-            SubscribeBuildEvents.Initialize(
+            DTE dte = (DTE)await GetServiceAsync(typeof(DTE));
+            SubscribeEvents.Initialize(
                 dte,
                 this);
-        }
 
+
+        }
+        protected override int QueryClose(out bool canClose)
+        {
+            SubscribeEvents.OnSessionEnd();
+            canClose = true; 
```

Comment:
If I remember correctly, setting `canClose` to true will prevent Visual Studio from shutting down.

Reply:
dit was standaard en heb ik niet aangepast, dus ik laat het op true. (als in, de methode overridden had dit al)